File: src/main/kotlin/ym_cosmetic/pick_perfume_be/perfume/entity/PerfumeDesigner.kt
package ym_cosmetic.pick_perfume_be.perfume.entity

import jakarta.persistence.*
import ym_cosmetic.pick_perfume_be.common.BaseTimeEntity
import ym_cosmetic.pick_perfume_be.designer.entity.Designer
import ym_cosmetic.pick_perfume_be.perfume.enums.DesignerRole

@Entity
@Table(name = "perfume_designer")
class PerfumeDesigner private constructor(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
        name = "perfume_id",
        nullable = false,
        foreignKey = ForeignKey(value = ConstraintMode.NO_CONSTRAINT)
    )
    val perfume: Perfume,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(
        name = "designer_id",
        nullable = false,
        foreignKey = ForeignKey(value = ConstraintMode.NO_CONSTRAINT)
    )
    val designer: Designer,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val designerRole: DesignerRole,

    @Column(length = 500)
    var content: String? = null
) : BaseTimeEntity() {

    companion object {
        fun create(
            perfume: Perfume,
            designer: Designer,
            role: DesignerRole,
            content: String? = null
        ): PerfumeDesigner {
            return PerfumeDesigner(
                perfume = perfume,
                designer = designer,
                designerRole = role,
                content = content
            )
        }
    }

    fun updatecontent(content: String?): PerfumeDesigner {
        this.content = content
        return this
    }
}

File: perfume-importer/src/entities/PerfumeDesigner.ts
import { Entity, ManyToOne, Enum, type EnumType } from "@mikro-orm/core";
import { BaseEntity } from "./BaseEntity";
import { Perfume } from "./Perfume";
import { Designer } from "./Designer";

export enum DesignerRole {
  PERFUMER = "perfumer",
  CREATIVE_DIRECTOR = "creative_director",
  NOSE = "nose",
}

@Entity()
export class PerfumeDesigner extends BaseEntity {
  @ManyToOne(() => Perfume)
  perfume!: Perfume;

  @ManyToOne(() => Designer)
  designer!: Designer;

  @Enum({ items: () => DesignerRole, type: "string" })
  role: DesignerRole = DesignerRole.PERFUMER;

  constructor(
    perfume: Perfume,
    designer: Designer,
    role: DesignerRole = DesignerRole.PERFUMER
  ) {
    super();
    this.perfume = perfume;
    this.designer = designer;
    this.role = role;
  }
}
